python/hsfs/storage_connector.py

```diff
@@ -14,38 +14,197 @@
+        return self._description
+
+    @abstractmethod
+    def spark_options(self):
+        pass
+
+
+class HopsFSConnector(StorageConnector):
+    type = StorageConnector.HOPSFS
 
     def __init__(
         self,
         id,
         name,
         featurestore_id,
-        storage_connector_type,
         description=None,
         # members specific to type of connector
         hopsfs_path=None,
         dataset_name=None,
+    ):
+        super().__init__(id, name, description, featurestore_id)
+
+        # HopsFS
+        self._hopsfs_path = hopsfs_path
+        self._dataset_name = dataset_name
+
+    def spark_options(self):
+        """Return prepared options to be passed to Spark, based on the additional
+        arguments.
+        """
+        return {}
+
+
+class S3Connector(StorageConnector):
+    type = StorageConnector.S3
+
+    def __init__(
+        self,
+        id,
+        name,
+        featurestore_id,
+        description=None,
+        # members specific to type of connector
         access_key=None,
         secret_key=None,
         server_encryption_algorithm=None,
         server_encryption_key=None,
         bucket=None,
+        session_token=None,
+        iam_role=None,
+    ):
+        super().__init__(id, name, description, featurestore_id)
+
+        # S3
+        self._access_key = access_key
+        self._secret_key = secret_key
+        self._server_encryption_algorithm = server_encryption_algorithm
+        self._server_encryption_key = server_encryption_key
+        self._bucket = bucket
+        self._session_token = session_token
+        self._iam_role = iam_role
+
+    @property
+    def access_key(self):
+        """Access key."""
+        return self._access_key
+
+    @property
+    def secret_key(self):
+        """Secret key."""
+        return self._secret_key
+
+    @property
+    def server_encryption_algorithm(self):
+        """Encryption algorithm if server-side S3 bucket encryption is enabled."""
+        return self._server_encryption_algorithm
+
+    @property
+    def server_encryption_key(self):
+        """Encryption key if server-side S3 bucket encryption is enabled."""
+        return self._server_encryption_key
+
+    @property
+    def bucket(self):
+        """Return the bucket for S3 connectors."""
+        return self._bucket
+
+    @property
+    def session_token(self):
+        """Session token."""
+        return self._session_token
+
+    @property
+    def iam_role(self):
+        """IAM role."""
+        return self._iam_role
+
+    @property
+    def path(self):
+        """If the connector refers to a path (e.g. S3) - return the path of the connector"""
+        return "s3://" + self._bucket
+
+    def spark_options(self):
+        """Return prepared options to be passed to Spark, based on the additional
+        arguments.
+        """
+        return {}
+
+    def prepare_spark(self, path: Optional[str] = None):
```

Comment:
where is this call being used?

Reply:
This is a convenience method, not used anywhere. I know you are the opinion everything should go through On demand feature groups eventually.